### PR TITLE
BNH-99 Profile pictures

### PR DESCRIPTION
**Pull request by MaxTheComputerer**
- Users' profile pictures from Google are stored in the database as URLs
- These are automatically fetched and updated
- If a landlord is linked to a user and the user has a profile picture, then this is displayed on the profile page, otherwise a default is used

Screenshot shows a landlord's profile picture on their profile page, and the current admin's picture in the navbar
image

image

### CODE REVIEW

File: web/Auth/ClaimsTransformer.cs [hunk 1]
```diff
@@ -32,14 +32,20 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
...
             // If they don't exist yet, create them
-            databaseUser = await CreateUserInDatabase(googleAccountId, existingClaimsIdentity);
-            _logger.LogInformation("Created new user '{UserEmail}' in database with id {UserId}", databaseUser.GoogleEmail, databaseUser.Id);
+            databaseUser = await CreateUserInDatabase(googleAccountId, googlePhotoUrl, existingClaimsIdentity);
+            _logger.LogInformation("Created new user '{UserEmail}' in database with id {UserId}",
+                databaseUser.GoogleEmail, databaseUser.Id);
+        }
+        else
+        {
+            await UpdateUserProfilePhoto(databaseUser, googlePhotoUrl);
```
Comment (oskarwilliams378) [inline web/Auth/ClaimsTransformer.cs:48]: Is this in order to update existing users?

File: web/Views/Admin/AdminDashboard.cshtml [hunk 2]
```diff
@@ -1,4 +1,6 @@
-﻿@model AdminDashboardViewModel
+﻿@using BricksAndHearts.Auth
+@using Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (oskarwilliams378) [inline web/Views/Admin/AdminDashboard.cshtml:2]: Do we need this? The _ViewImports is importing this :)

**Review by oskarwilliams378 — CHANGES_REQUESTED** [review]
Very minor comments

File: web/Auth/ClaimsTransformer.cs (hunk 1 above)
Comment (MaxTheComputerer) [reply web/Auth/ClaimsTransformer.cs:48]: Yes!

File: web/Views/Admin/AdminDashboard.cshtml (hunk 2 above)
Comment (MaxTheComputerer) [reply web/Views/Admin/AdminDashboard.cshtml]: Yeah it seems we don't need it, I think Rider automatically imports it

File: web/Auth/ClaimsTransformer.cs (hunk 1 above)
Comment (oskarwilliams378) [reply web/Auth/ClaimsTransformer.cs:48]: could you make the else be an else if instead then, checking that if that database user already has that googlePhotoUrl?

Comment (MaxTheComputerer) [reply web/Auth/ClaimsTransformer.cs:48]: Ah yeah good idea!

**Review by oskarwilliams378 — APPROVED** [review]

**Review by cec200wire — APPROVED** [review]

**Pull request merged**